core/src/panic.rs

```diff
@@ -7,7 +8,23 @@ pub fn configure_panic() {
         // Invoke the default hook and exit the process
         default_hook(panic_info);
         println!("Exiting...");
-        // TODO: setup a wait time and fold the log system properly
+
+        // Get the panic location and message
+        let (file, line) = match panic_info.location() {
+            Some(location) => (location.file(), location.line()),
+            None => ("unknown", 0),
+        };
+
+        let message = match panic_info.payload().downcast_ref::<&str>() {
+            Some(s) => *s,
+            None => match panic_info.payload().downcast_ref::<String>() {
+                Some(s) => &s[..],
+                None => "unknown",
+            },
+        };
+        // Log the panic
+        error!("Panic at {}:{}: {}", file, line, message);
```

Comment:
`PanicInfo` implements `Display` almost the same way as yours. Could you check how the  hook work with `panic_any` passing custom type?

There's method `mesaage` , however it's only accessable on nightly toolchain. Although `std` uses it when implements `Display`

I would suggest to use `Display` fmt if it works with panic_any, otherwise there's no difference to me